metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_query.py

```diff
@@ -880,7 +880,7 @@ def table_upstreams_with_column_lineage(
             on qid.query_id = query_history.query_id
         )
         SELECT
-            h.downstream_table_name AS "DOWNSTREAM_TABLE_NAME",
+            REGEXP_REPLACE(h.downstream_table_name, '__DBT_TMP', '') AS "DOWNSTREAM_TABLE_NAME",
```

Comment:
Hey @john-claro-cko , I've two suggestions here:

1. Let's use `REPLACE` rather than `REGEX_REPLACE` here.
2. let's move this to earlier in the query to make sure the new value is used all other places(aggregations). I'm thinking - you can move this to line 776 -> `REPLACE(w.value : "objectName" :: varchar, '__DBT_TMP', '') AS downstream_table_name,`

Reply:
also quick check - do you have dbt ingestion set up?  Doesn't it already ingest the lineage you are looking for ?

Reply:
Thanks Mayuri, done it now. Do you think we should also do it for upstream tables?

And, yes, we do have DBT ingestions setup but the lineage was not rendering at all.

Reply:
Currently the understanding is that lineage would be best generated by ETL tool's (in this case - dbt) ingestion itself rather than snowflake, as snowflake ingestion does not handle multi-hop lineage via temporary staging tables which is usually how most ETL tools do it. You can start a thread about missing lineage in dbt ingestion in #ingestion and @hsheth2 will be able to help on that.

For upstream tables, we currently omit all temporary staging tables generated by known ETL tools from upstream in query using deny filter, so this would not work / would not be needed for upstream tables.

Reply:
Makes sense, I'll log something with a more descriptive thread. Happy to keep it on`downstream_table_name` for now then.